### ISSUE TITLE
Getting MemoryError on line 161 of hunspell/hunspell.pyx

### ISSUE THREAD

**Issue by joshuairl** [OpenGov/cython_hunspell#5]
Any ideas what is causing this?

I'm just calling `h.spell('word')`.

What does this generally mean?

Running on python 3.5.2, latest version of cyhunspell

**Comment by MSeal** [comment]
Hmm that's odd, that means `<bytes>py_string.encode(encoding, 'strict')` returned a length <= 0 bytes array.

What OS are you running on? Are there any other resource constraints or odd things about your setup? Where you for example possibly out of memory?

It's likely something specific to your machine as I'm not able to reproduce naively.
```
Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from hunspell import Hunspell
>>> h = Hunspell()
>>> h.spell('word')
True
```

**Comment by MSeal** [comment]
Going to close as there's been no response for a week and nothing I can reproduce atm

**Comment by SophieStar** [comment]
@MSeal I'm having the same problem as described above. I'm using the german dictionary from Open Office.

My code looks like this, the error occurs when calling `hobj.spell(word)`:
```
hobj = Hunspell('de_DE_OLDSPELL', hunspell_data_dir=de_dic, disk_cache_dir=cache_dir)
for word in sentence.split(' '):
  isCorrect = hobj.spell(word)
  if isCorrect is True:
    correctedWord = word
  else:
    suggestions = hobj.suggest(word)
```

And this is my log output (the first 4000 + n words are corrected correctly):
```
2017-09-14 22:57:56.869027 2000 Words corrected.
2017-09-14 22:57:56.875385 2500 Words corrected.
2017-09-14 22:57:56.880535 3000 Words corrected.
2017-09-14 22:57:56.886838 3500 Words corrected.
2017-09-14 22:57:56.890703 4000 Words corrected.
Traceback (most recent call last):
  File "...../main.py", line 37, in <module>
    text_array_spell_corrected_hun = spell_corrector.replaceMissspelledWordsWithHunSpell(text_array)
  File "...../lib/spell_check.py", line 123, in replaceMissspelledWordsWithHunSpell
    isCorrect = hobj.spell(word)
  File "hunspell\hunspell.pyx", line 161, in hunspell.hunspell.HunspellWrap.spell (hunspell/hunspell.cpp:3905)
MemoryError
```

**Comment by MSeal** [comment]
Can you upload the sentence that you're parsing? I should be able to debug this with that, without this gives a little more context to explore.

**Comment by MSeal** [comment]
@SophieStar I think I found the issue -- there's a bug where empty strings cause the program to think there was a memory allocation error. I'll post a fix this weekend, if you could confirm it fixed your issue afterwards I'd appreciate it :)

**Comment by MSeal** [comment]
Issue should be resolved with 1.2.1 release. Let me know if you run into Memory errors again (and add the sentences causing the issue if it does crop up).
